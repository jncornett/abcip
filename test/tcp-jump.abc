a(syn)
b(syn, ack)
a(ack)

a(ack, len=100)
a(ack, len=100, jump=+100)
a(ack, len=100, jump=-200)
a(ack, len=100, jump=+100)

a(fin,ack)
b(fin,ack)
a(ack)

