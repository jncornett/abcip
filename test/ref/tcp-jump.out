IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [S], cksum 0xb3d9 (correct), seq 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0001 0000 4006 5cbc 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0008 0000 0001 0000 0000 5002  ..............P.
	0x0030:  2000 b3d9 0000                           ......
IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.8 > 10.1.2.3.48620: Flags [S.], cksum 0xb3c7 (correct), seq 1, ack 2, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0001 0000 4006 5cbc 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0008 bdec 0000 0001 0000 0002 5012  ..............P.
	0x0030:  2000 b3c7 0000                           ......
IP (tos 0x0, ttl 64, id 2, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0xb3c8 (correct), ack 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0002 0000 4006 5cbb 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0008 0000 0002 0000 0002 5010  ..............P.
	0x0030:  2000 b3c8 0000                           ......
IP (tos 0x0, ttl 64, id 3, offset 0, flags [none], proto TCP (6), length 140)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0x9fe2 (correct), seq 1:101, ack 1, win 8192, length 100
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  008c 0003 0000 4006 5c56 0a01 0203 0a09  ......@.\V......
	0x0020:  0807 bdec 0008 0000 0002 0000 0002 5010  ..............P.
	0x0030:  2000 9fe2 0000 4c6f 7265 6d20 6970 7375  ......Lorem.ipsu
	0x0040:  6d20 646f 6c6f 7220 7369 7420 616d 6574  m.dolor.sit.amet
	0x0050:  2c20 636f 6e73 6563 7465 7475 7220 6164  ,.consectetur.ad
	0x0060:  6970 6973 6369 6e67 2065 6c69 742e 2045  ipiscing.elit..E
	0x0070:  7469 616d 2073 7573 6369 7069 742c 2071  tiam.suscipit,.q
	0x0080:  7561 6d20 6174 2073 6365 6c65 7269 7371  uam.at.scelerisq
	0x0090:  7565 2075 6c74 7269 6369                 ue.ultrici
IP (tos 0x0, ttl 64, id 4, offset 0, flags [none], proto TCP (6), length 140)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0x15ba (correct), seq 201:301, ack 1, win 8192, length 100
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  008c 0004 0000 4006 5c55 0a01 0203 0a09  ......@.\U......
	0x0020:  0807 bdec 0008 0000 00ca 0000 0002 5010  ..............P.
	0x0030:  2000 15ba 0000 7320 6469 7320 7061 7274  ......s.dis.part
	0x0040:  7572 6965 6e74 206d 6f6e 7465 732c 206e  urient.montes,.n
	0x0050:  6173 6365 7475 7220 7269 6469 6375 6c75  ascetur.ridiculu
	0x0060:  7320 6d75 732e 204e 756c 6c61 6d20 6461  s.mus..Nullam.da
	0x0070:  7069 6275 7320 7269 7375 7320 696e 2064  pibus.risus.in.d
	0x0080:  6f6c 6f72 2069 6163 756c 6973 2071 7569  olor.iaculis.qui
	0x0090:  7320 6c61 6369 6e69 6120                 s.lacinia.
IP (tos 0x0, ttl 64, id 5, offset 0, flags [none], proto TCP (6), length 140)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0xb874 (correct), seq 101:201, ack 1, win 8192, length 100
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  008c 0005 0000 4006 5c54 0a01 0203 0a09  ......@.\T......
	0x0020:  0807 bdec 0008 0000 0066 0000 0002 5010  .........f....P.
	0x0030:  2000 b874 0000 6573 2c20 6e69 7369 206e  ...t..es,.nisi.n
	0x0040:  756c 6c61 2066 6572 6d65 6e74 756d 206f  ulla.fermentum.o
	0x0050:  7263 692c 2061 2069 6d70 6572 6469 6574  rci,.a.imperdiet
	0x0060:  206c 6163 7573 2065 6e69 6d20 6120 6665  .lacus.enim.a.fe
	0x0070:  6c69 732e 2043 756d 2073 6f63 6969 7320  lis..Cum.sociis.
	0x0080:  6e61 746f 7175 6520 7065 6e61 7469 6275  natoque.penatibu
	0x0090:  7320 6574 206d 6167 6e69                 s.et.magni
IP (tos 0x0, ttl 64, id 6, offset 0, flags [none], proto TCP (6), length 140)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0x2ece (correct), seq 301:401, ack 1, win 8192, length 100
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  008c 0006 0000 4006 5c53 0a01 0203 0a09  ......@.\S......
	0x0020:  0807 bdec 0008 0000 012e 0000 0002 5010  ..............P.
	0x0030:  2000 2ece 0000 6573 7420 7065 6c6c 656e  ......est.pellen
	0x0040:  7465 7371 7565 2e20 4d61 6563 656e 6173  tesque..Maecenas
	0x0050:  2075 7420 6665 6c69 7320 6573 742c 2073  .ut.felis.est,.s
	0x0060:  6564 2069 6e74 6572 6475 6d20 6c6f 7265  ed.interdum.lore
	0x0070:  6d2e 2043 7261 7320 6175 6374 6f72 2c20  m..Cras.auctor,.
	0x0080:  6970 7375 6d20 7669 7461 6520 7665 6e65  ipsum.vitae.vene
	0x0090:  6e61 7469 7320 6661 7563                 natis.fauc
IP (tos 0x0, ttl 64, id 7, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [F.], cksum 0xb237 (correct), seq 401, ack 1, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0007 0000 4006 5cb6 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0008 0000 0192 0000 0002 5011  ..............P.
	0x0030:  2000 b237 0000                           ...7..
IP (tos 0x0, ttl 64, id 2, offset 0, flags [none], proto TCP (6), length 40)
    10.9.8.7.8 > 10.1.2.3.48620: Flags [F.], cksum 0xb236 (correct), seq 1, ack 402, win 8192, length 0
	0x0000:  0201 0203 0405 0209 0807 0605 0800 4500  ..............E.
	0x0010:  0028 0002 0000 4006 5cbb 0a09 0807 0a01  .(....@.\.......
	0x0020:  0203 0008 bdec 0000 0002 0000 0193 5011  ..............P.
	0x0030:  2000 b236 0000                           ...6..
IP (tos 0x0, ttl 64, id 8, offset 0, flags [none], proto TCP (6), length 40)
    10.1.2.3.48620 > 10.9.8.7.8: Flags [.], cksum 0xb236 (correct), ack 2, win 8192, length 0
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0028 0008 0000 4006 5cb5 0a01 0203 0a09  .(....@.\.......
	0x0020:  0807 bdec 0008 0000 0193 0000 0003 5010  ..............P.
	0x0030:  2000 b236 0000                           ...6..
